code/DDSCodeTester.cpp

```diff
@@ -5693,21 +5722,65 @@ int main(
         int argc,
         const char** argv)
 {
-    if (argc != 6)
-    {
-        printf("Bad number of parameters\n");
-        exit(-1);
-    }
-
     // Also show log warnings to spot potential mistakes
     Log::SetVerbosity(Log::Kind::Warning);
     // Report filename and line number for debugging
     Log::ReportFilenames(true);
 
     int exit_code = 0;
-    if (!dds_permissions_test(argv[1], argv[2], argv[3], argv[4], argv[5]))
+    if (argc == 6)
+    {
+        if (!dds_permissions_test(argv[1], argv[2], argv[3], argv[4], argv[5]))
+        {
+            std::cout << "Error parsing persimission xml file" << std::endl;
+            exit_code = -1;
+        }
+    }
+    else if (argc == 2)
     {
-        std::cout << "Error parsing persimission xml file" << std::endl;
+        if (strncmp(argv[1], "Static", 6) == 0)
+        {
+            std::string file = argv[1];
+            DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+            if (factory->check_xml_static_discovery(file) != ReturnCode_t::RETCODE_OK)
+            {
+                printf("1Error parsing xml file %s\n", argv[1]);
+                exit_code = -1;
+            }
+
+            std::string fileData = "data://<?xml version=\"1.0\" encoding=\"utf-8\"?>" \
+                    "<staticdiscovery>" \
+                    "<participant>" \
+                    "<name>HelloWorldPublisher</name>" \
+                    "<writer>" \
+                    "<userId>1</userId>" \
+                    "<entityID>2</entityID>" \
+                    "<topicName>HelloWorldTopic</topicName>" \
+                    "<topicDataType>HelloWorld</topicDataType>" \
+                    "</writer>" \
+                    "</participant>" \
+                    "</staticdiscovery>";
+            if (factory->check_xml_static_discovery(fileData) != ReturnCode_t::RETCODE_OK)
+            {
+                printf("2Error parsing xml file %s\n", argv[1]);
```

Comment:
```suggestion
                printf("Error parsing xml file %s\n", argv[1]);
```

Or is there a reason to keep those numbers?